### PR TITLE
Add RBi for matrix

### PR DESCRIPTION
**Pull request by Morriar**
No signature

### CODE REVIEW

**Comment by Morriar** [comment]
^ Fixed tests

**Comment by jez** [comment]
https://go/builds/bui_HMLzSeSa6iw0Ej

File: rbi/stdlib/matrix.rbi [hunk 1]
```diff
@@ -0,0 +1,577 @@
...
+  # Also aliased as:
+  # [`cross`](https://docs.ruby-lang.org/en/2.6.0/Vector.html#method-i-cross)
+  def cross_product(*vs); end
+
+  # Alias for:
+  # [`inner_product`](https://docs.ruby-lang.org/en/2.6.0/Vector.html#method-i-inner_product)
+  def dot(v); end
+
+  # Iterate over the elements of this vector
+  def each(&block); end
```
Comment (jez) [inline rbi/stdlib/matrix.rbi:211]: https://docs.ruby-lang.org/en/2.6.0/Matrix.html#method-i-each

`each` is missing a positional param. the autogenerated signatures we have for matrix at stripe flag this as an incompatible redefinition:


```
sorbet/shims/autogenerated/gems/matrix.rbi:247: Method Matrix#each redefined without matching argument count. Expected: 0, got: 1 http://go/e/4010
     247 |  def each(which=T.unsafe(nil), &blk); end
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    https://github.com/sorbet/sorbet/tree/0983722275df15f4e2f102c872358508a4e6c497/external/com_stripe_ruby_typer/rbi/stdlib/matrix.rbi#L345: Previous definition
     345 |  def each(&block); end
            ^^^^^^^^^^^^^^^^
Errors: 1
```

File: rbi/stdlib/matrix.rbi [hunk 2]
```diff
@@ -0,0 +1,577 @@
...
+end
+
+# The `Matrix` class represents a mathematical matrix. It provides methods for
+# creating matrices, operating on them arithmetically and algebraically, and
+# determining their mathematical properties such as trace, rank, inverse,
+# determinant, or eigensystem.
+class Matrix
+  include ::Enumerable
+
+  Elem = type_member(:out)
```
Comment (gmalette) [inline rbi/stdlib/matrix.rbi:324]: I think this makes `Matrix` unusable in a typed file. 

- `T::Matrix` is not defined
- `def Matrix.[]` is defined by Ruby

Ex: in `sig { params(m: Matrix).void }` Sorbet-static complains that `Matrix` is a generic without type argument. While `Matrix[Integer]` satisfies the static part, it fails at Runtime.

**Pull request merged**